File: ruby/reverse_string.rb
require "pry"

def reverse_string(str)
  arr = str.split("")
  newArr = []
  for i in 0..arr.length-1
    newArr.unshift(arr[i]) 
  end
  newArr.join("")
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  let arr = str.split("")
  let newArr = []
  for(let i = arr.length-1;i >= 0; i--){
    newArr.push(arr[i]) 
  }
  return newArr.join("")
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
